### ISSUE TITLE
Rqt plugin for monitoring (like rqt_graph)

### ISSUE THREAD

**Issue by stonier** [robotics-in-concert/rocon_multimaster#119]
Groovy shadow-fixed is broken, and fuerte doesn't have many components.

Just wait this one out.

**Comment by stonier** [comment]
Implemented over several commits in rocon_gateway_graph, finalising with 006ed7c21036247dca89924d070a73cf3ffceb21.

It's not got alot of detail yet, to do that we really need to subclass the NodeItem object and implement things like hover callbacks.

**Comment by ablasdel** [comment]
Was testing plugins for groovy and I ran into this error when opening the gateway graph.

Would it be better to open a separate issue?

```
RosPluginProvider.load(rocon_gateway_graph/Gateway Graph) exception raised in __builtin__.__import__(rocon_gateway_graph.gateway_graph, [GatewayGraph]):
Traceback (most recent call last):
  File "/u/ablasdel/groovy_sandbox/rqt/rqt_gui/src/rqt_gui/ros_plugin_provider.py", line 78, in load
    module = __builtin__.__import__(attributes['module_name'], fromlist=[attributes['class_from_class_type']], level=0)
  File "/opt/ros/groovy/stacks/rocon_multimaster/rocon_gateway_graph/src/rocon_gateway_graph/gateway_graph.py", line 34, in <module>
    from rocon_gateway import Graph
  File "/opt/ros/groovy/stacks/rocon_multimaster/rocon_gateway/src/rocon_gateway/__init__.py", line 19, in <module>
    from .hub_api import resolve_hub
  File "/opt/ros/groovy/stacks/rocon_multimaster/rocon_gateway/src/rocon_gateway/hub_api.py", line 10, in <module>
    import redis
ImportError: No module named redis

PluginManager._load_plugin() could not load plugin "rocon_gateway_graph/Gateway Graph":
Traceback (most recent call last):
  File "/opt/ros/groovy/lib/python2.7/dist-packages/qt_gui/plugin_handler.py", line 89, in load
    self._load()
  File "/opt/ros/groovy/lib/python2.7/dist-packages/qt_gui/plugin_handler_direct.py", line 54, in _load
    self._plugin = self._plugin_provider.load(self._instance_id.plugin_id, self._context)
  File "/opt/ros/groovy/lib/python2.7/dist-packages/qt_gui/composite_plugin_provider.py", line 71, in load
    instance = plugin_provider.load(plugin_id, plugin_context)
  File "/opt/ros/groovy/lib/python2.7/dist-packages/qt_gui/composite_plugin_provider.py", line 71, in load
    instance = plugin_provider.load(plugin_id, plugin_context)
  File "/u/ablasdel/groovy_sandbox/rqt/rqt_gui_py/src/rqt_gui_py/ros_py_plugin_provider.py", line 68, in load
    return super(RosPyPluginProvider, self).load(plugin_id, plugin_context)
  File "/opt/ros/groovy/lib/python2.7/dist-packages/qt_gui/composite_plugin_provider.py", line 71, in load
    instance = plugin_provider.load(plugin_id, plugin_context)
  File "/u/ablasdel/groovy_sandbox/rqt/rqt_gui/src/rqt_gui/ros_plugin_provider.py", line 84, in load
    raise e
ImportError: No module named redis

```

ros-groovy-rocon-multimaster 0.1.5-s1355148058~precise
ros-groovy-rqt 0.1.5-s1355146052~precise

**Comment by stonier** [comment]
We just updated today as well, some to latest groovy beta and some to groovy shadow-fixed.

It seems there's an import problem that's crept into ros with the rosbuilds. I'll write here as I get updates.

**Comment by stonier** [comment]
Error seems to be coming from the change in https://github.com/ros/rospkg/issues/14. If you manually edit that back to the original line for now, that should be fine (so long as you don't have fuerte catkin packages I suspect). Hopefully Dirk will pick up on this quickly.

**Comment by ablasdel** [comment]
We were looking at the redis package and it seems it has a catkin tag in there.
That is what is causing the issue.

**Comment by stonier** [comment]
Ach - went 90% of the way to finding the problem and missed it. Thanks Aaron.

Fixed in bd84bba2ef8fa2f2e52da8640c149df8124f0f83

I'll rerelease.

**Comment by stonier** [comment]
Remote flip and gateway edges added in #130. 

The layout is not so great though - check if it can be spread out a bit more than it is.
